### PR TITLE
Optimizations to Length and Normalize in Vector3 and Vector4

### PR DESCRIPTION
**Pull request by nkast**
Length() and LengthSquared() was calcualated through
DistanceSquared(this, zero).

### CODE REVIEW

**Comment by nkast** [comment]
Also note that in Vector2 the sqLength calculation is inlined for Length() & Normalize()
and Normalize() code is sightly different between Vector3 & Vector4.

File: MonoGame.Framework/Vector3.cs [hunk 1]
```diff
@@ -568,7 +568,7 @@ public static void Hermite(ref Vector3 value1, ref Vector3 tangent1, ref Vector3
         /// <returns>The length of this <see cref="Vector3"/>.</returns>
         public float Length()
         {
-            float result = DistanceSquared(this, zero);
+            float result = this.LengthSquared();
```
Comment (tomspilman) [inline MonoGame.Framework/Vector3.cs]: I say go ahead and inline the length squared calculation here.

File: MonoGame.Framework/Vector3.cs [hunk 2]
```diff
@@ -791,7 +791,7 @@ public void Normalize()
         /// <returns>Unit vector.</returns>
         public static Vector3 Normalize(Vector3 value)
         {
-            float factor = Distance(value, zero);
+            float factor = value.Length();
```
Comment (tomspilman) [inline MonoGame.Framework/Vector3.cs]: Go ahead and inline these too like in Vector2.

**Comment by tomspilman** [comment]
Good catch @nkast .  This performance bug has been in MG since the first commit to GitHub 7 years ago.

**Comment by tomspilman** [comment]
Thanks @nkast !

**Pull request merged**